ElementX/Sources/Services/Timeline/TimelineProxy.swift

```diff
@@ -631,4 +631,21 @@ extension Array where Element == TimelineItemProxy {
         
         return nil
     }
+    
+    func firstEventTimelineItemUsingEventOrTransactionID(_ eventOrTransactionID: EventOrTransactionId) -> EventTimelineItem? {
+        for item in self {
+            if case let .event(eventTimelineItem) = item {
+                switch eventTimelineItem.id {
+                case .event(_, let identifier):
+                    if identifier == eventOrTransactionID {
+                        return eventTimelineItem.item
+                    }
+                default:
+                    break
+                }
+            }
```

Comment:
This might be slightly cleaner:
```suggestion
            if case let .event(eventTimelineItem) = item,
               case let .event(_, identifier) = eventTimelineItem.id,
               identifier == eventOrTransactionID {
                return eventTimelineItem.item
            }
```

Reply:
Yup yup, nice, thank you!